src/cargo-kani/src/metadata.rs

```diff
@@ -138,6 +139,14 @@ impl KaniSession {
             Ok(metadata.proof_harnesses.clone())
         }
     }
+
+    /// Deserialize a *.restrictions.json file
+    fn read_kani_metadata(&self, path: &Path) -> Result<KaniMetadata> {
```

Comment:
Why do you need `self` here?

Reply:
I thought this was a function that would be useful in other files as well, so i made it a utililty method for the current Session.

Reply:
1. I think we should only parse the file once.
2. Can you create a public utility function instead that doesn't rely on self?